content/wasm-functions/using-cron-jobs.md

```diff
@@ -0,0 +1,227 @@
+
+Let's modify the response body to validate that our Spin application is running on a specific interval, and determine whether it was triggered by our cron job or via a direct request. If the request includes query parameters, we can infer that it came from the cron job, which we will set up in future steps. We’ll use this identifier to log the cron event. To apply these changes, navigate to the `hello-cron-job` directory, go to `src/index.js`, and add the following code snippet:
+
+<!-- @selectiveCpy -->
+
+```javascript 
+// For AutoRouter documentation refer to https://itty.dev/itty-router/routers/autorouter
+import { AutoRouter } from 'itty-router';
+
+// Initialize the router
+let router = AutoRouter();
+
+// Route ordering matters:
+// - The first matching route is used
+// - Routes can act as middleware if they don’t return a response
+// - Unmatched routes return a 404
+router
+    .get("/cron-me", (request) => {
+        // Parse the request URL to access query parameters
+        const url = new URL(request.url);
+        
+        // Read the 'msg' query parameter, defaulting to an empty string if not provided
+        const msg = url.searchParams.get("msg");
+
+        // Capture current timestamp for logging and response
+        const now = new Date().toISOString();
+
+        // Log the trigger event, including the value of 'msg'
+        if (msg) {
+            console.log(`Hello, cron job triggered with msg: "${msg}" at ${now}`);
+            // Return a success response with a status 200
+            return new Response( {
+                status: 200,
+            });
+        } else {
+            // Return a direct ping response without logging
+            return new Response(`Hello! This is a direct ping at ${now}`, {
+                status: 200,
+            });
+        }
+    });
+
+// Attach the router to the fetch event
+addEventListener('fetch', (event) => {
+    event.respondWith(router.fetch(event.request));
+});
+```
+
+To build and run the application locally, run the following command:
+
+<!-- @selectiveCpy -->
+
+```console
+spin build --up
+```
+
+Let's curl the application's endpoint to test if it is working as expected:
+
+<!-- @selectiveCpy -->
+
+```console
+curl localhost:3000/cron-me
+```
+
+ You should see a message returned along the lines of (_but with an accurate datetime at the time of following this tutoral_):
+
+<!-- @nocpy -->
+
+```console
+Hello! This is a direct ping at 2025-04-02T02:45:25.562Z
+```
+
+## Deploy Spin Application to Fermyon Wasm Functions
+
+Before scheduling our cron job, we deploy our Spin application to the Fermyon Wasm Functions platform. Deploy the application with the following command:
+
+<!-- @selectiveCpy -->
+
+```console
+spin aka deploy
+```
+
+Upon successful deployment, you should receive a domain name that you can use to test the application:
```

Comment:
Consider shoiwing the deploy output so they can see where to find the domain name.